packages/base/field-component.gts

```diff
@@ -204,6 +212,37 @@ function fieldsComponentsFor<T extends BaseDef>(
         // don't handle symbols or nulls
         return Reflect.get(target, property, received);
       }
+
+      let stable: BoxComponent | undefined;
+
+      // debugger;
```

Comment:
comment that can be removed

Reply:
Yeah sorry this last commit was actually more of a WIP from pairing today with @habdelra, I'll move this PR back to draft as there are still some unresolved issues where we'll need @ef4's input - mainly around [this test failure](https://github.com/cardstack/boxel/actions/runs/7023107399/job/19108857887?pr=860#step:8:2373) and how to make the new cache mechanism work with changes in the array